[14.0] Add edi_exchange_type_auto

simahawk opened the pull request:
Ease automation of EDI exchange types within the EDI framework.

Moved from https://github.com/OCA/edi/pull/663

Depends on:
- [x] https://github.com/OCA/edi/pull/778
- [x] https://github.com/OCA/edi/pull/797

simahawk committed:
Add edi_exchange_type_auto

simahawk committed:
edi_exchange_type_auto: make _edi_auto_log_skip flexible

simahawk committed:
edi_exchange_type_auto: fix tests

simahawk committed:
edi_exchange_type_auto: skip with flag too

simahawk committed:
edi_exchange_type_auto: fix logging

simahawk committed:
edi_exchange_type_auto: fix handling multiple records

simahawk committed:
edi_exchange_type_auto: fix skip via model conf

simahawk committed:
edi_exchange_type_auto: add more tests

simahawk committed:
edi_exchange_type_auto: skip generate for disabled partners

simahawk committed:
edi_exc_type_auto: allow to use snippet for conf

simahawk committed:
edi_exc_type_auto: simplify _edi_auto_collect_todo

simahawk committed:
edi_exchange_type_auto: use type rules

simahawk committed:
edi_exc_type_auto: fix do not generate if already generated

yvaucher commented on edi_exchange_type_auto/readme/DESCRIPTION.rst:
```diff
@@ -0,0 +1 @@
+TODO
```
Don't forget the readme parts ;)

simahawk committed:
edi_exchange_type_auto: fix generate

Avoid duplicating records
Improve job result msg
Post msg only when needed

Works better w/ quick_exec

simahawk committed:
edi_exchange_type_auto: fix logger name

simahawk committed:
edi_exchange_type_auto: create new record if former is pending

If the record has not been sent out yet for whatever reason
(job delayed, job failed, send failed, etc)
we still want to generate a new up to date record to be sent.
